qiskit_nature/second_q/operators/electronic_integrals.py

```diff
@@ -0,0 +1,313 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""ElectronicIntegrals class"""
+
+from __future__ import annotations
+
+from copy import deepcopy
+from typing import Iterator, cast
+
+import numpy as np
+
+from qiskit.quantum_info.operators.mixins import LinearMixin
+
+from qiskit_nature.exceptions import QiskitNatureError
+from qiskit_nature.settings import settings
+
+from .polynomial_tensor import PolynomialTensor
+from .tensor_ordering import (
+    IndexType,
+    _chem_to_phys,
+    find_index_order,
+)
+
+
+class ElectronicIntegrals(LinearMixin):
+    """ElectronicIntegrals class"""
+
+    def __init__(
+        self,
+        alpha: PolynomialTensor | None = None,
+        beta: PolynomialTensor | None = None,
+        mixed: PolynomialTensor | None = None,
+    ) -> None:
+        """TODO."""
+        self._register_length = alpha.register_length if alpha is not None else None
+        self._data: dict[str, PolynomialTensor] = {}
+        self.alpha = alpha
+        self.beta = beta
+        self.mixed = mixed
```

Comment:
Actually, this structure is reminiscent of a previous implementation prototype where `ElectronicIntegrals` was trying to implement a Composite-pattern around the `PolynomialTensor`. This is no longer the case (and also not wanted) so this can be changed to simply be three `PolynomialTensor`-type attributes.